src/class/api.php

```diff
@@ -0,0 +1,356 @@
+        // To prevent notices.
+        $_PE = array_pad($_PE, 5, '');
+
+        // $_PE[0] should always be API.
+        if (strtolower($_PE[0]) != 'api') {
+            $this->aResponse['errors'][] = 'Could not parse requested URL.';
+            $this->sendHeader(400, true); // Send 400 Bad Request, print response, and quit.
+        }
+
+        // Check if we're using the old style LOVD2-style API or not.
+        if (in_array($_PE[1], array('rest', 'rest.php'))) {
+            // Yes, we are...
+            $this->aResponse['version'] = 0;
+            // This API also ignores the Accept header.
+            $this->sFormatOutput = 'text/plain';
+            // And, we only allow GET.
+            if (!GET) {
+                // Will only allow GET.
+                // $this->aResponse['errors'][] = 'Method not allowed here.';
+                // $this->sendHeader(405, true); // Send 405 Method Not Allowed, print response, and quit.
+                // This API is LOVD2-style and shouldn't have their output changed now that we're more advanced.
+                header('HTTP/1.0 501 Not Implemented');
+                exit;
+            }
+
+            // Parse URL to see what we need to do.
+            list(,,
+                $this->sResource, // 2
+                $this->sGene,     // 3
+                $this->nID) = $_PE;
+
+            if (!$this->sResource) { // No data type given.
+                header('HTTP/1.0 400 Bad Request');
+                die('Too few parameters.');
+            } elseif (!in_array($this->sResource, array('variants', 'genes'))) { // Wrong data type given.
+                header('HTTP/1.0 400 Bad Request');
+                die('Requested data type not known.');
+            } elseif ($this->sResource == 'variants' && !$this->sGene) { // Variants, but no gene selected.
+                header('HTTP/1.0 400 Bad Request');
+                die('Too few parameters.');
+            }
+
+        } else {
+            // This is the new LOVD3-style API.
+            // URLs can be with or without version.
+            $aURLElements = $_PE;
+            array_shift($aURLElements); // We take "api" off.
+
+            if (preg_match('/^v([0-9]+)$/', $aURLElements[0], $aRegs)) {
+                // We received version in URL.
+                // Version must be larger than zero, and no greater than the currently configured version.
+                if (!$aRegs[1] || $aRegs[1] > $this->nVersion) {
+                    $this->aResponse['errors'][] = 'Requested version is unavailable.';
+                    $this->sendHeader(400, true); // Send 400 Bad Request, print response, and quit.
+                } else {
+                    $this->nVersion = (int) $aRegs[1];
+                }
+                array_shift($aURLElements); // Take the version off.
+            }
+
+            // Next, should be resource.
+            $this->sResource = array_shift($aURLElements);
```

Comment:
There's an "else" in between there, so those lines of code are never both run :wink: